go/vt/env/env.go

```diff
@@ -72,11 +76,15 @@ func VtMysqlRoot() (string, error) {
 		return root, nil
 	}
 
-	// otherwise let's look for mysqld in the PATH.
-	// ensure that /usr/sbin is included, as it might not be by default
-	// This is the default location for mysqld from packages.
-	newPath := fmt.Sprintf("/usr/sbin:%s", os.Getenv("PATH"))
-	os.Setenv("PATH", newPath)
+	// Otherwise let's look for mysqld in the PATH env var.
+	// Ensure that /usr/sbin is included, as it might not be by default
+	// and this is often the default location used by mysqld system
+	// packages (apt, dnf, etc).
+	once.Do(func() {
+		newPath := fmt.Sprintf("/usr/sbin:%s", os.Getenv("PATH"))
```

Comment:
nit: we _could_ parse the path to check if `/usr/sbin` is included and add it only if it is not. May or may not be worth the additional code / processing given that we only ever do this once per process execution.

Reply:
Yeah, I thought about that but given how often this is called it felt better to do this once rather than perform the string search every time.

Reply:
On the other hand... it's possible that the PATH gets changed elsewhere and then we later DO NOT have it in the PATH. So maybe I should check for it and add it if needed...

Reply:
For the safety/determinism reason noted, I went ahead and changed it: https://github.com/vitessio/vitess/pull/15096/commits/e90e849ae6440605f88d5f020df294c26fa89f4d